enhancements/installer/openstack-ipi-failure-domains.md

```diff
@@ -0,0 +1,342 @@
+- Validate the latency between the nodes. We expect all the availability zones to be located in the same datacenter. The latency between the nodes must remain acceptable for etcd to function correctly.
+
+## Proposal
+
+The Installer allows users to provide a list of Failure Domains which contain information about networking and storage. The Failure domains is defined as an OpenStack platform scoped property. Failure domains are referenced in OpenStack MachinePools.
+
+The Installer validates the information given by the user and return an error if a resource doesn't exist or is incorrectly used.
+
+The cluster will be deployed in these domains on day 1 via IPI.
+
+Nodes will be provisioned in the different Failure Domains in a Round Robin fashion.
+
+When using Failure Domains, the users will have to do it for both the control plane and the computes machine pools.
+
+The infrastructure resources owned by the cluster continue to be clearly identifiable and distinct from other resources.
+
+Destroying a cluster must make sure that no resources are deleted that didn't belong to the cluster.  Leaking resources is preferred over any possibility of deleting non-cluster resources.
+
+### Workflow Description
+
+- The OpenStack administrators will deploy OpenStack with multiple Availability Zones, so Nova and Cinder can respectfully deploy servers and volumes in the Failure Domain.
+- The OpenStack administrators will create at least one Routed Provider Network, and then multiple subnets, where each zone has a least one subnet.
+- The OpenShift administrators will identify the Failure Domains: their Availability Zones, subnets, etc. They'll decide which ones will be used for the masters, and the ones for the computes (failure domains can be shared).
+- The OpenShift administrators will provide the right configuration in the `install-config.yaml` file and then deploy the cluster (detailed later in this document).
+- (Out of scope for this enhancement, see "Non-Goals") The OpenShift administrators will make sure that the network infrastructure has a networking route to reach the OCP Control Plane VIPs.
+
+### API Extensions
+
+The Cloud team is currently working on adding support for [Control Plane Machine Set](https://github.com/openshift/enhancements/blob/master/enhancements/machine-api/control-plane-machine-set.md).
+It is possible we'll want to update the Control Plane Machine Set API to match the OpenStack failure domains defined here.
+
+
+### Implementation Details/Notes/Constraints
+
+#### Types
+
+```golang
+package openstack
+
+// Platform stores all the global configuration that all
+// machinesets use.
+type Platform struct {
+    ...
+
+    // FailureDomains configures failure domain information for the OpenStack platform
+    // +optional
+    FailureDomains *[]FailureDomain `json:"failureDomains,omitempty"`
```

Comment:
Are you sure we can put that in `Platform`? This has to be used from `func tfVars`...